lib/foreman/observable.rb

```diff
@@ -0,0 +1,19 @@
+module Foreman
+  module Observable
+    def trigger_hook(hook_name, payload: nil, &blk)
+      payload = payload_for(payload, blk)
+
+      do_notify_event_observers(hook_name, payload: payload)
+    end
+
+    private
+
+    def payload_for(payload, blk)
+      payload || blk&.call(self)
+    end
+
+    def do_notify_event_observers(event_name, payload:)
+      ActiveSupport::Notifications.instrument(event_name, payload)
```

Comment:
I think it would make sense to add `foreman` namespace so we can filter only events sent from our API by using `/^foreman./` regex
```suggestion
      ActiveSupport::Notifications.instrument("foreman.#{event_name}", payload)
```

Reply:
Yes, please do.

Reply:
actually, to follow the rails naming convention which is `event.library` we should add `foreman` namespace as a suffix